client/src/main/java/io/pravega/client/byteStream/impl/BufferedByteStreamWriterImpl.java

```diff
@@ -97,6 +98,12 @@ public void flush() throws IOException {
         out.flush();
     }
 
+    @Override
+    public CompletableFuture<Void> flushAsync() throws IOException {
+        commitBuffer();
```

Comment:
This violates the locking contract set up above. 
The contract of `@GuardedBy("$lock")` specifies that every access to the buffer is guarded by the lock. However here the method is not `@Synchronized` and as `commitBuffer` is not either no lock is held in this path when the buffer is modified.

Reply:
made the method @Synchronous